### ISSUE TITLE
Merge libpcp into the libnatpmp 

### ISSUE THREAD

**Issue by stokito** [miniupnp/libnatpmp#24]
When searching for a PCP library most users get to the [libpcp](https://github.com/libpcp/pcp/) which is kind of a reference implementation. The library has IPv6 support. But it's not published in Linux distros and as far I understood is not widelly used. But it still receives some PRs. Having two libraries that do the same thing confuses users.

Is it possible to merge the PCP code from the libpcp into the libnatpmp?
I asked their authors https://github.com/libpcp/pcp/issues/22 but maybe libnatpmp can do that.
Maybe if you are ok but don't have a time to do this then please mark the issue as "help wanted".

**Comment by miniupnp** [comment]
What do you mean by merging ?
why not use libpcp ?

**Comment by masterflitzer** [comment]
isn't pcp the successor to nat-pmp? they're not the same thing or am i missing something?

**Comment by stokito** [comment]
Yes, the PCP is rework of the NATPMP made by Apple for their routers.
But they don't make new routers so the protocol seems abandoned.
Almost all products that I saw just use the libnatpmp.
I didn't learned it deeply so actually I don't sure even if the PCP has some really big advantages over the NATPMP. Maybe we should just forget it to avoid fragmentation and supporting burden.
I may be wrong because not a specialist in the area.

**Comment by masterflitzer** [comment]
natpmp is by apple and pcp was standardized 2013 as successor, idk if i understood your comment correctly but to clarify: pcp should be used and we should forget natpmp to avoid fragmentation

the other way around makes no sense and would be supporting old standards where newer ones are at our fingertips
